tests/storage/cbt/pull_collect_runner.py

```
@@ -0,0 +1,222 @@
+
+# Duplicated in push_restore_runner.py — both runners are standalone scripts executed
+# inside pods and cannot share imports from the test framework.
+def _checkpoint_timestamp_from_path(path: str, checkpoint_timestamp_pattern: str) -> str:
+    """Return the checkpoint timestamp embedded in a backup path."""
+    match = re.search(checkpoint_timestamp_pattern, path)
+    return match.group(1) if match else ""
+
+
+def _list_client_backup_raw_files(params: dict[str, Any]) -> list[str]:
+    """Return sorted raw backup files from pull-mode client storage."""
+    result = subprocess.run(
```

Ruff (0.15.21)

[error] `subprocess` call: check for execution of untrusted input

(S603)